test/rekt/features/apiserversource/data_plane.go

```diff
@@ -822,6 +823,73 @@ func SendsEventsWithRetries() *feature.Feature {
 	return f
 }
 
+func setupNodeLabels(nodeLabels map[string]string) feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		nodes, err := client.Get(ctx).CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Could not list nodes: %v", err)
+		}
+
+		if len(nodes.Items) == 0 {
+			t.Fatal("No nodes found")
+		}
+
+		randomNode := &nodes.Items[0]
+
+		randomNode.Labels = nodeLabels
+
+		// Update the node with the new labels
+		_, err = client.Get(ctx).CoreV1().Nodes().Update(ctx, randomNode, metav1.UpdateOptions{})
+
+		if err != nil {
+			t.Fatalf("Could not update node: %v", err)
+		}
+	}
+}
+
+func DeployAPIServerSauceWithNodeSelector() *feature.Feature {
+	f := feature.NewFeatureNamed("deploy")
+
+	// same labels as in the configmap
+
+	f.Requirement("setup config-features and load into context", func(ctx context.Context, t feature.T) {
+		env := environment.FromContext(ctx)
+		ns := env.Namespace()
+		configmap.Install("config-features", ns)
+	})
+	f.Requirement("setup node labels", setupNodeLabels(apiserversource.TestLabels()))
+
+	source := feature.MakeRandomK8sName("apiserversource")
+	sink := feature.MakeRandomK8sName("sink")
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+
+	sacmName := feature.MakeRandomK8sName("apiserversource")
+	f.Setup("Create Service Account for ApiServerSource with RBAC for sources.knative.dev/v1 PingSources",
+		setupAccountAndRoleForPingSources(sacmName))
+
+	cfg := []manifest.CfgFn{
+		apiserversource.WithServiceAccountName(sacmName),
+		apiserversource.WithEventMode("Reference"),
+		apiserversource.WithSink(service.AsDestinationRef(sink)),
+		apiserversource.WithResources(v1.APIVersionKindSelector{
+			APIVersion: "sources.knative.dev/v1",
+			Kind:       "PingSource",
+		}),
+		apiserversource.WithNamespaceSelector(&metav1.LabelSelector{
+			MatchLabels:      map[string]string{},
+			MatchExpressions: []metav1.LabelSelectorRequirement{},
+		}),
+	}
+
+	f.Setup("install ApiServerSource", apiserversource.Install(source, cfg...))
+	f.Setup("ApiServerSource goes ready", apiserversource.IsReady(source))
```

Comment:
```suggestion
	f.Requirement("install ApiServerSource", apiserversource.Install(source, cfg...))
	f.Requirement("ApiServerSource goes ready", apiserversource.IsReady(source))
```
This will make sure you install the apiserversource *after* all the setup stages are completed